src/components/atoms/BaseCard/BaseCard.tsx

```diff
@@ -67,17 +96,68 @@ export const BaseCard = forwardRef<CardHandle, BaseCardProps>(
       flip: (forceFaceUp?: boolean) => turnCard(forceFaceUp),
     }));
 
-    const onClick = () => !disableNativeEvents && turnCard();
+    const onClick = () => !disableNativeEvents && !isDragging && Date.now() - startDate < 200 && turnCard();
+
     const onKeyDown = (event: KeyboardEvent<HTMLDivElement>) =>
       !disableNativeEvents && event.keyCode === 32 && turnCard();
 
+    const handleMouseDown = useCallback(
+      ({ clientX, clientY }) => {
+        if (!disableNativeEvents) {
+          setStartDate(Date.now());
+          setStartPosition({ x: clientX, y: clientY });
+          setIsDraging(true);
+        }
+      },
+      [disableNativeEvents]
+    );
+
+    const handleMouseMove = useCallback(
+      ({ clientX, clientY }) => {
+        const newCardPosition: Position = {
+          x: Math.round(cardPosition.x + clientX - startPosition.x),
+          y: Math.round(cardPosition.y + clientY - startPosition.y),
+        };
+        setCardPosition(newCardPosition);
+      },
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      [startPosition]
+    );
+
+    const handleMouseUp = useCallback(() => {
+      setIsDraging(false);
```

Comment:
This actually goes a little bit with what I was talking on the onClick one.
Can't be the `onClick` that does that instead of turning the card then?